sympy/solvers/solveset.py

```diff
@@ -314,6 +311,48 @@ def _invert_complex(f, g_ys, symbol):
     return (f, g_ys)
 
 
+def _invert_abs(f, g_ys, symbol):
+    """Helper function for inverting absolute value functions.
+
+    Returns the complete result of inverting an absolute value
+    function along with the conditions which must also be satisfied.
+
+    If it is certain that all these conditions are met, a `FiniteSet`
+    of all possible solutions is returned. If any condition cannot be
+    satisfied, an `EmptySet` is returned. Otherwise, a `ConditionSet`
+    of the solutions, with all the required conditions specified, is
+    returned.
+
+    """
+
+    n = Dummy('n', real=True)
+
+    if f.is_positive:
+        f_n = f
+    else:
+        f_n = f.args[0]
+
+    g_x, values = _invert_real(f_n, Union(imageset(Lambda(n, n), g_ys),
+                               imageset(Lambda(n, -n), g_ys)), symbol)
+
+    if not isinstance(g_ys, FiniteSet):
+        raise NotImplementedError
+    else:
+        unknown = []
+        for a in g_ys.args:
+            ok = a.is_nonnegative if a.is_Number else a.is_positive
+            if ok is None:
+                unknown.append(a)
+            elif not ok:
+                return g_x, S.EmptySet
+        if unknown:
+            conditions = And(*[Contains(i, Interval(0, oo))
+                for i in unknown])
+        else:
+            conditions = True
+    return g_x, ConditionSet(g_x, conditions, values)
+
```

Comment:
Can we use the following, it might slight reduce computation

```diff

diff --git a/sympy/solvers/solveset.py b/sympy/solvers/solveset.py
index 08c6253..980bf3c 100644
--- a/sympy/solvers/solveset.py
+++ b/sympy/solvers/solveset.py
@@ -338,18 +338,13 @@ def _invert_abs(f, g_ys, symbol):
     if not isinstance(g_ys, FiniteSet):
         raise NotImplementedError
     else:
-        unknown = []
+        conditions = True
         for a in g_ys.args:
             ok = a.is_nonnegative if a.is_Number else a.is_positive
             if ok is None:
-                unknown.append(a)
+                conditions = And(*[Contains(a, Interval(0, oo))])
             elif not ok:
                 return g_x, S.EmptySet
-        if unknown:
-            conditions = And(*[Contains(i, Interval(0, oo))
-                for i in unknown])
-        else:
-            conditions = True
     return g_x, ConditionSet(g_x, conditions, values)

```

Reply:
I'm not sure that that will be faster. Do you have some experience that says it will? Typically, I try to avoid incremental updates to a return value (e.g. collect args then return `Add(*args)` instead of doing `rv += arg` in a loop).

Reply:
Since two loops need to be traversed, time complexity may slightly increase to `O(m+n)` (where m is len(g_ys.args), n is len(unknown)), whereas the above would lead to `O(m)`. This won't be of much difference.
I suggested because why to unnecessary traverse another loop when this can be achieved with one, and also it might reduce some code.

Reply:
@Yathartha22 , here is an example of what I mean:
```
>>> timeit('''
... args = []
... for i in arg:
...   args.append(i)
... And(*args)
... ''', '''from sympy import var, And\narg = var('a:z')''', number=100)
0.23920276681502628
>>> timeit('''
... rv = True
... for i in arg:
...   rv = And(rv, i)
... ''', '''from sympy import var, And\narg = var('a:z')''', number=100)
0.8529278468695338
```

Reply:
Yes @smichr I also checked with and interestingly it seems that the one I suggested would take longer. Any reason why the approach takes longer time even when updation takes place in parallel.